liger_iris_pipeline/tests/test_preprocess_flatfield.py

```diff
@@ -1,82 +1,29 @@
-#!/usr/bin/env python
-# coding: utf-8
-
-# In[ ]:
-
-
+# Imports
 import liger_iris_pipeline
-
-
-# In[ ]:
-
-
-# check where are we loading the library from
-liger_iris_pipeline.__file__
-
-
-# In[ ]:
-
-
+liger_iris_pipeline.monkeypatch_jwst_datamodels()
 import numpy as np
-
-
-# In[ ]:
-
-
-from test_utils import get_data_from_url
-
-
-# In[ ]:
-
-
+from astropy.io import fits
+from liger_iris_pipeline.tests.test_utils import get_data_from_url
 from jwst import datamodels
 
+def test_process_flatfield():
 
-# In[ ]:
-
-
-raw_science_filename = get_data_from_url("17903858")
-
-
-# In[ ]:
-
-
-input_model = datamodels.open(raw_science_filename)
-
-
-# In[ ]:
-
-
-input_model
-
-
-# In[ ]:
-
-
-step = liger_iris_pipeline.pipeline.ProcessFlatfieldL2()
-
-
-# In[ ]:
-
-
-step_output = step.run(raw_science_filename)
-
-
-# In[ ]:
-
-
-dark_current = datamodels.open(step.dark_current.dark_name)
-
-
-# In[ ]:
-
-
-expected = input_model.data - dark_current.data
-expected /= np.median(expected)
+    # Grab flat field
+    raw_flat_filename = get_data_from_url("48191521")
+    input_model = datamodels.open(raw_flat_filename)
 
+    # Initialize flatfield pipeline
+    pipeline = liger_iris_pipeline.pipeline.ProcessFlatfieldL2()
 
-# In[ ]:
+    # Run pipeline
+    pipeline_output = pipeline.run(raw_flat_filename)
 
+    # Open dark
+    dark_current = datamodels.open(pipeline.dark_current.dark_name)
 
-np.testing.assert_allclose(step_output[0].data, expected)
+    # Manually create a dark subtracted master flat
+    expected = input_model.data - dark_current.data
+    expected /= np.median(expected)
 
+    # Test
+    np.testing.assert_allclose(pipeline_output[0].data, expected)
```

Comment:
I don't remember why we are only checking `[0]`, can you add a comment here?

Reply:
I actually wasn't sure how it worked before without the `[0]`. The `Pipeline` produces this list:

https://github.com/oirlab/liger_iris_pipeline/blob/437e564aec0e958f5b070f87d1fdc641fe2e464d/liger_iris_pipeline/pipeline/preprocess_flatfield.py#L56

Maybe the general philosophy is that a pipeline can output multiple data products?

Reply:
yes, it can have multiple input dataset in the input association